contracts/router/quoter/PoolQuoterV1.sol

```diff
@@ -0,0 +1,398 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IDefaultPoolCalc} from "../interfaces/IDefaultPoolCalc.sol";
+import {IDefaultExtendedPool} from "../interfaces/IDefaultExtendedPool.sol";
+import {ILinkedPool} from "../interfaces/ILinkedPool.sol";
+import {IPausable} from "../interfaces/IPausable.sol";
+import {ISwapQuoterV1, PoolToken, SwapQuery} from "../interfaces/ISwapQuoterV1.sol";
+import {Action, DefaultParams} from "../libs/Structs.sol";
+import {UniversalTokenLib} from "../libs/UniversalToken.sol";
+
+/// @notice Stateless abstraction to calculate exact quotes for any DefaultPool instances.
+abstract contract PoolQuoterV1 is ISwapQuoterV1 {
+    IDefaultPoolCalc internal immutable _defaultPoolCalc;
+    address internal immutable _weth;
+
+    constructor(address defaultPoolCalc, address weth) {
+        _defaultPoolCalc = IDefaultPoolCalc(defaultPoolCalc);
```

Comment:
Is there an example of what the `defaultPoolCalc` implementation will be?

Reply:
Yep, this is going to be #255!

Reply:
Resolved in ef0ee5afa32922a0b83e46df754fc0ad0e82835f